make deps fails on Vagrant ubuntu14 due to gmake error

It fails on [provision.sh:136](https://github.com/facebook/osquery/blob/master/tools/provision.sh#L136) with the following error:

```
$ cmake -DCMAKE_CXX_FLAGS=-fPIC -DGFLAGS_NAMESPACE:STRING=google .
-- The CXX compiler identification is Clang 3.4.0
-- Check for working CXX compiler: /usr/bin/clang++
CMake Error: Generator: execution of make failed. Make command was: /usr/bin/gmake "cmTryCompileExec304280116/fast"
-- Check for working CXX compiler: /usr/bin/clang++ -- broken
CMake Error at /usr/share/cmake-2.8/Modules/CMakeTestCXXCompiler.cmake:54 (message):
  The C++ compiler "/usr/bin/clang++" is not able to compile a simple test
  program.

  It fails with the following output:

   Change Dir: /vagrant/.sources/gflags-2.1.1/CMakeFiles/CMakeTmp



  Run Build Command:/usr/bin/gmake "cmTryCompileExec304280116/fast"

  No such file or directory

  Generator: execution of make failed.  Make command was: /usr/bin/gmake
  "cmTryCompileExec304280116/fast"





  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:16 (project)


-- Configuring incomplete, errors occurred!
See also "/vagrant/.sources/gflags-2.1.1/CMakeFiles/CMakeOutput.log".
See also "/vagrant/.sources/gflags-2.1.1/CMakeFiles/CMakeError.log".
```

This is solved by `sudo ln -s /usr/bin/make /usr/bin/gmake`, which we should probably just do somewhere earlier in the build process if `which gmake` fails.

Investigating whether this occurs on other OSes.

--- a/tools/provision.sh
+++ b/tools/provision.sh
@@ -132,6 +132,9 @@ function install_gflags() {
     if [[ ! -d gflags-2.1.1 ]]; then
       tar -xf v2.1.1.tar.gz
     fi
+    if [[ ! -x "$(which gmake)" ]]; then
+      sudo ln -s `which make` /usr/local/bin/gmake
+    fi
     pushd gflags-2.1.1
     cmake -DCMAKE_CXX_FLAGS=-fPIC -DGFLAGS_NAMESPACE:STRING=google .
     make

